monitoring/histogram.cc

```diff
@@ -58,17 +58,22 @@ size_t HistogramBucketMapper::IndexForValue(const uint64_t value) const {
 }
 
 namespace {
-  const HistogramBucketMapper bucketMapper;
+
+const HistogramBucketMapper& BucketMapper() {
+  static const auto* const kBucketMapper = new HistogramBucketMapper{};
```

Comment:
Isn't this a memory leak?

Reply:
In the sense that the destructor for the single static `HistogramBucketMapper` object created here will never be called, and the memory allocated here will never be reclaimed.

This is actually one of the recommended paradigms for safe global initialization, namely: Safe Initialization, No Destruction. See [here](go/totw/110#the-fix-safe-initialization-no-destruction), and we do this safely in a few places in FBKV already (like [here](https://github.com/FirebasePrivate/fbkv/blob/9a24f0d167f0c1b9f351ef51b6a1912adaed70e5/storage/bson/util.h#L70))

Reply:
OK, apparently it's not: https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables
Also it turns out that since c++11 this is thread safe as well.

Reply:
I just didn't know the static initialization line is skipped for all but the first invocation :-). Neat.

BTW. Great totw